File: java/Easy/ParkingSystem.java
package Easy;

public class ParkingSystem {

    int[] pspace = new int[4];
    
    public ParkingSystem(int big, int medium, int small) {
        pspace[1] = big;
        pspace[2] = medium;
        pspace[3] = small;
    }
    
    public boolean addCar(int carType){
        if(pspace[carType] > 0){
            pspace[carType]--;
            return true;
        } 
        return false;
    }
}

class TstPark {

    public static void main(String[] args){
        ParkingSystem obj = new ParkingSystem(2, 2, 1);
        boolean p1 = obj.addCar(2);
        boolean p2 = obj.addCar(2);
        boolean p3 = obj.addCar(2);
        System.out.println(p1);
        System.out.println(p2);
        System.out.println(p3);
    }
}

File: python/Easy/Parkingsystem.py
class ParkingSystem:
    a = [0, 0, 0, 0]
    def __init__(self, big: int, medium: int, small: int):
        self.a[1] = big
        self.a[2] = medium
        self.a[3] = small


    def addCar(self, carType: int) -> bool:
        if self.a[carType] > 0:
            self.a[carType] -= 1
            return True
        return False
    
s = ParkingSystem(2, 2, 2)
print(s.addCar(2))
print(s.addCar(2))
print(s.addCar(2))
    
